app/controllers/api/v1/admin/students/playlist_activities_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Api
+  module V1
+    module Admin
+      module Students
+        class PlaylistActivitiesController < Api::ApiController
+          respond_to :json
+          before_action :set_student
+
+          def update
+            @student.playlist_activities.each do |activity|
+              activity.update(position: new_position(activity.id))
+            end
+            respond_with @student
+          end
+
+          private
+
+            # def authorize_resource
+            #   playlist_activity = Student.find(params[:student_id]).playlist_activities.first
```

Comment:
student authorization as it stands only allows a teacher admin to edit them, I assume teachers without admin ability should be able to edit their playlist. The playlist activity currently allows any teacher to edit, that's why we added the id to the end of the route so we can check the authorization with params[:id] for the playlist activity in the authorize resource method...even though we are doing a batch edit. 

Further, the route as it stands does provide the ability to edit via an external service.